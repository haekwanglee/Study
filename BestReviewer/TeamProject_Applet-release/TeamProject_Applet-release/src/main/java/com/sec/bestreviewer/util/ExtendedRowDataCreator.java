package com.sec.bestreviewer.util;

import com.sec.bestreviewer.database.Row;
import com.sec.bestreviewer.database.field.EmployeeSchema;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.util.LinkedList;
import java.util.List;

public class ExtendedRowDataCreator {
    private static final String NAME_DELIMITER = " ";
    private static final String PHONE_NUM_DELIMITER = "-";
    private static final String DATE_PATTERN = "yyyyMMdd";

    private static final int FIRST_NAME_INDEX = 0;
    private static final int LAST_NAME_INDEX = 1;

    private static final int MIDDLE_NUMBER_INDEX = 1;
    private static final int LAST_NUMBER_INDEX = 2;

    public static Row create(Row row) {
        final List<String> updatedDataList = new LinkedList<>(row.getRow());

        appendNames(row, updatedDataList);
        appendPhoneNumbers(row, updatedDataList);
        appendDates(row, updatedDataList);

        return new Row(updatedDataList);
    }

    private static void appendNames(Row row, List<String> extendedList) {
        final String name = row.getValue(EmployeeSchema.NAME.getIndex());
        final String[] splitName = name.split(NAME_DELIMITER);
        extendedList.add(splitName[FIRST_NAME_INDEX]);
        extendedList.add(splitName[LAST_NAME_INDEX]);
    }

    private static void appendPhoneNumbers(Row row, List<String> updatedDataList) {
        final String phoneNumber = row.getValue(EmployeeSchema.PHONE_NUMBER.getIndex());
        final String[] splitPhoneNumber = phoneNumber.split(PHONE_NUM_DELIMITER);
        updatedDataList.add(splitPhoneNumber[MIDDLE_NUMBER_INDEX]);
        updatedDataList.add(splitPhoneNumber[LAST_NUMBER_INDEX]);
    }

    private static void appendDates(Row row, List<String> updatedDataList) {
        final String birthDay = row.getValue(EmployeeSchema.BIRTHDAY.getIndex());
        LocalDate date = LocalDate.parse(birthDay, DateTimeFormatter.ofPattern(DATE_PATTERN));
        updatedDataList.add(String.valueOf(date.getYear()));
        updatedDataList.add(getDateString(date.getMonthValue()));
        updatedDataList.add(getDateString(date.getDayOfMonth()));
    }

    private static String getDateString(int monthOrDay) {
        String date = String.valueOf(monthOrDay);
        return date.length() == 1
                ? "0" + date
                : date;
    }
}
