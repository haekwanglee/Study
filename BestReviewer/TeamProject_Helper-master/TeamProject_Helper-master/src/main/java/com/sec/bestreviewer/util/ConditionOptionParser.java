package com.sec.bestreviewer.util;

public class ConditionOptionParser {
    private String secondOption = " ";
    private String thirdOption = " ";

    public String getSecondOption() {
        return secondOption;
    }
    public String getThirdOption() {
        return thirdOption;
    }

    public void setSecondOption(String secondOption) {
        this.secondOption = secondOption;
    }

    public void setThirdOption(String thirdOption) {
        this.thirdOption = thirdOption;
    }
}
