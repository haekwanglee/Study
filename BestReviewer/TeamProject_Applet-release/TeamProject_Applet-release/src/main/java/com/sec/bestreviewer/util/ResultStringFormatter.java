package com.sec.bestreviewer.util;

import com.sec.bestreviewer.database.Row;
import com.sec.bestreviewer.database.field.EmployeeSchema;

import java.util.Comparator;
import java.util.List;
import java.util.stream.Collectors;

public class ResultStringFormatter {
    private static final int YEAR_1990_EMPLOYEE_NUMBER = 90_000000;
    private static final int YEAR_PREFIX_19 = 1900_000000;
    private static final int YEAR_PREFIX_20 = 2000_000000;

    public static List<String> getEmployeeListToFormattedString(List<Row> employeeList) {
        return employeeList.stream()
                .sorted(Comparator.comparing(ResultStringFormatter::getEmployeeNumber))
                .map(ResultStringFormatter::getEmployeeToFormattedString)
                .collect(Collectors.toList());
    }

    private static Integer getEmployeeNumber(Row employee) {
        int employeeNumber = Integer.parseInt(employee.getValue(EmployeeSchema.getPrimaryKeyIndex()));
        return employeeNumber >= YEAR_1990_EMPLOYEE_NUMBER ?
                YEAR_PREFIX_19 + employeeNumber : YEAR_PREFIX_20 + employeeNumber;
    }

    public static String getEmployeeToFormattedString(Row employee) {
        return employee.getValue(EmployeeSchema.EMPLOYEE_NUMBER.getIndex()) + "," +
                employee.getValue(EmployeeSchema.NAME.getIndex()) + "," +
                employee.getValue(EmployeeSchema.CAREER_LEVEL.getIndex()) + "," +
                employee.getValue(EmployeeSchema.PHONE_NUMBER.getIndex()) + "," +
                employee.getValue(EmployeeSchema.BIRTHDAY.getIndex()) + "," +
                employee.getValue(EmployeeSchema.CERTI.getIndex());

    }
}