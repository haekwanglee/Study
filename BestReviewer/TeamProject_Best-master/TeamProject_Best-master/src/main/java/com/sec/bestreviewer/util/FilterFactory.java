package com.sec.bestreviewer.util;

import com.sec.bestreviewer.store.filter.AndFilter;
import com.sec.bestreviewer.store.filter.Filter;
import com.sec.bestreviewer.store.filter.OrFilter;
import com.sec.bestreviewer.store.filter.SingleFilter;

import java.util.List;

import static com.sec.bestreviewer.util.ParamsParser.PairIndex.FIRST_PAIR;
import static com.sec.bestreviewer.util.ParamsParser.PairIndex.SECOND_PAIR;

public class FilterFactory {

    public static Filter buildFilter(List<String> params, OptionParser optionParser) {
        if (optionParser.getAndOption().isAndOn()) {
            Pair<String, String> conditionPair1 = ParamsParser.parser(params, FIRST_PAIR);
            OptionParser optionParser1 = new OptionParser(optionParser.getAndOption().getFirstOptions(), conditionPair1.fieldName);

            Pair<String, String> conditionPair2 = ParamsParser.parser(params, SECOND_PAIR);
            OptionParser optionParser2 = new OptionParser(optionParser.getAndOption().getSecondOptions(), conditionPair2.fieldName);
            return new AndFilter(new SingleFilter(conditionPair1, optionParser1), new SingleFilter(conditionPair2, optionParser2));
        }

        if (optionParser.getOrOption().isOrOn()) {
            Pair<String, String> conditionPair1 = ParamsParser.parser(params, FIRST_PAIR);
            OptionParser optionParser1 = new OptionParser(optionParser.getOrOption().getFirstOptions(), conditionPair1.fieldName);

            Pair<String, String> conditionPair2 = ParamsParser.parser(params, SECOND_PAIR);
            OptionParser optionParser2 = new OptionParser(optionParser.getOrOption().getSecondOptions(), conditionPair2.fieldName);

            return new OrFilter(new SingleFilter(conditionPair1, optionParser1), new SingleFilter(conditionPair2, optionParser2));
        }

        Pair<String, String> conditionPair = ParamsParser.parser(params, FIRST_PAIR);
        return new SingleFilter(conditionPair, optionParser);
    }
}
