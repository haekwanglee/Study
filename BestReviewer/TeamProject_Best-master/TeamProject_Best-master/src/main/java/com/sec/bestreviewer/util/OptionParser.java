package com.sec.bestreviewer.util;

import com.sec.bestreviewer.option.*;

import java.util.Arrays;
import java.util.List;

public class OptionParser {

    public final static List<String> SUPPORTED_OPTIONS = Arrays.asList("-p", "-f", "-l", "-m", "-y", "-d", "-g", "-ge", "-s", "-se", "-a", "-o");
    private final PrintOption printOption;
    private final NameOption nameOption;
    private final BirthDayOption birthDayOption;
    private final PhoneOption phoneOption;
    private final EqualityOption equalityOption;
    private final AndOption andOption;
    private final OrOption orOption;

    public OptionParser(List<String> options, String fieldName) {
        printOption = new PrintOption(options);
        nameOption = new NameOption(options, fieldName);
        birthDayOption = new BirthDayOption(options, fieldName);
        phoneOption = new PhoneOption(options, fieldName);
        equalityOption = new EqualityOption(options);
        andOption = new AndOption(options);
        orOption = new OrOption(options);
    }

    public PrintOption getPrintOption() {
        return printOption;
    }

    public NameOption getNameOption() {
        return nameOption;
    }

    public BirthDayOption getBirthDayOption() {
        return birthDayOption;
    }

    public PhoneOption getPhoneOption() {
        return phoneOption;
    }

    public EqualityOption getEqualityOption() {
        return equalityOption;
    }

    public AndOption getAndOption() {
        return andOption;
    }

    public OrOption getOrOption() {
        return orOption;
    }
}
