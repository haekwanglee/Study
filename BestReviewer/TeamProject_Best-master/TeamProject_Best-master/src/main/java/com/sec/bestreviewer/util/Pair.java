package com.sec.bestreviewer.util;

import java.util.Objects;

public class Pair<F, S> {
    public final F fieldName;
    public final S value;

    private Pair(F fieldName, S value) {
        this.fieldName = fieldName;
        this.value = value;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Pair)) {
            return false;
        }
        Pair<?, ?> p = (Pair<?, ?>) o;
        return Objects.equals(p.fieldName, fieldName) && Objects.equals(p.value, value);
    }

    @Override
    public int hashCode() {
        return (fieldName == null ? 0 : fieldName.hashCode()) ^ (value == null ? 0 : value.hashCode());
    }

    @Override
    public String toString() {
        return "Pair{" + String.valueOf(fieldName) + " " + String.valueOf(value) + "}";
    }

    public static <A, B> Pair <A, B> create(A a, B b) {
        return new Pair<A, B>(a, b);
    }
}