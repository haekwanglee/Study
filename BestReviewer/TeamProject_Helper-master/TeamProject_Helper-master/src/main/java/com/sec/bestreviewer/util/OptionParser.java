package com.sec.bestreviewer.util;

import com.sec.bestreviewer.store.ConditionParam;

import java.util.List;

public class OptionParser {
    private final int INDEX_PRINT_OP = 0;
    private final int INDEX_MATCH_OP = 1;
    private final int INDEX_COMP_OP = 2;
    private final int INDEX_ANDOR_OP = 3;
    private final int INDEX_MATCH_OP_AFTER_ANDOR = 4;
    private final int INDEX_COMP_OP_AFTER_ANDOR = 5;

    private final int OPTION_LIST_MIN = 1;
    private final int OPTION_LIST_NORMAL = 3;

    private static final String OP_PRINT = "-p";
    private static final String OP_AND = "-a";
    private static final String OP_OR = "-o";

    private boolean isPrintOn;
    private boolean isAndOperationOn = false;
    private boolean isOrOperationOn = false;
    private Pair<ConditionOptionParser, ConditionOptionParser> conditionOptions;


    public OptionParser(List<String> options) {
        ConditionOptionParser conditionFirst = new ConditionOptionParser();
        ConditionOptionParser conditionSecond = new ConditionOptionParser();

        isPrintOn = OP_PRINT.equals(options.get(INDEX_PRINT_OP));

        if (options.size() > OPTION_LIST_MIN) {
            conditionFirst.setSecondOption(options.get(INDEX_MATCH_OP));
            conditionFirst.setThirdOption(options.get(INDEX_COMP_OP));
        }

        if (options.size() > OPTION_LIST_NORMAL) {
            isAndOperationOn = OP_AND.equals(options.get(INDEX_ANDOR_OP));
            isOrOperationOn = OP_OR.equals(options.get(INDEX_ANDOR_OP));
            conditionSecond.setSecondOption(options.get(INDEX_MATCH_OP_AFTER_ANDOR));
            conditionSecond.setThirdOption(options.get(INDEX_COMP_OP_AFTER_ANDOR));
        }

        conditionOptions = Pair.create(conditionFirst, conditionSecond);
    }

    public boolean isPrintOn() {
        return isPrintOn;
    }

    public boolean isAndOperation() { return isAndOperationOn; }

    public boolean isOrOperation() {
        return isOrOperationOn;
    }

    public String getSecondOption() {
        return conditionOptions.first.getSecondOption();
    }

    public String getThirdOption() {
        return conditionOptions.first.getThirdOption();
    }

    public String getSecondOptionAfterAndOr() {
        return conditionOptions.second.getSecondOption();
    }

    public String getThirdOptionAfterAndOr() {
        return conditionOptions.second.getThirdOption();
    }
}