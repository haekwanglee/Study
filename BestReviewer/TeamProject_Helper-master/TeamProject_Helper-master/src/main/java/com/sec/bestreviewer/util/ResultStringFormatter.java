package com.sec.bestreviewer.util;

import com.sec.bestreviewer.store.Employee;

import java.util.Collections;
import java.util.Comparator;
import java.util.List;
import java.util.stream.Collectors;

public class ResultStringFormatter {

    public static List<String> getEmployeeListToFormattedString(List<Employee> employeeList,
                                                                String commandType, int count) {
        if (employeeList == null || employeeList.size() == 0) {
            return Collections.singletonList(commandType + ",NONE");
        }
        return employeeList.stream()
                .sorted(Comparator.comparing(Employee::getEmployeeNumber))
                .limit(count)
                .map(employee -> getEmployeeToFormattedString(commandType, employee))
                .collect(Collectors.toList());
    }

    public static String getEmployeeToFormattedString(String commandType, Employee employee) {
        return  commandType + "," + employee.getEmployeeNumber() + "," +
                employee.getName() + "," +
                employee.getCareerLevel() + "," +
                employee.getPhoneNumber() + "," +
                employee.getBirthday() + "," +
                employee.getCerti();
    }

    public static List<String> getEmployeeListToFormattedString(List<Employee> employeeList, String commandType) {
        final String formattedString = employeeList.size() == 0 ? "NONE" : Integer.toString(employeeList.size());
        return Collections.singletonList(commandType + "," + formattedString);
    }
}