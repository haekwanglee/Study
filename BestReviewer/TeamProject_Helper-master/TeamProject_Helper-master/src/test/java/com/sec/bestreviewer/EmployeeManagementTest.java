package com.sec.bestreviewer;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;


import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.util.Random;

import static org.junit.jupiter.api.Assertions.assertTrue;

public class EmployeeManagementTest {
    @Test
    public void testArgumentsEmptyInputFile() throws Exception {
        File inputFile = File.createTempFile("empty", ".txt");
        String outputFileName = "./src/test/java/com/sec/bestreviewer/output.txt";
        String[] args = {inputFile.getAbsolutePath(), outputFileName};
        EmployeeManagement employeeManagement = new EmployeeManagement();
        employeeManagement.run(args);

        File file = new File(outputFileName);

        assertTrue(file.exists());
    }

    @Test
    public void testArgumentsWrongArgsCount() throws Exception {
        String[] args = {"input.txt"};

        EmployeeManagement employeeManagement = new EmployeeManagement();

        Assertions.assertThrows(IllegalArgumentException.class, () -> {
            employeeManagement.run(args);
        });
    }

    @Test
    public void testArgumentsNotExistInputFile() throws Exception {
        String[] args = {"notexist.txt", "output.txt"};

        EmployeeManagement employeeManagement = new EmployeeManagement();

        Assertions.assertThrows(IllegalArgumentException.class, () -> {
            employeeManagement.run(args);
        });
    }

    @Test
    public void testArgumentsWrongCommand() throws Exception {
        final String inputFileName = "./src/test/java/com/sec/bestreviewer/wrong_command_test_input.txt";
        final String outputFileName = "./src/test/java/com/sec/bestreviewer/wrong_command_test_output.txt";

        String[] args = {inputFileName, outputFileName};
        File outputFile = new File(outputFileName);

        EmployeeManagement employeeManagement = new EmployeeManagement();

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    public void testIntegration() throws Exception {
        final String outputFileName = "./src/test/java/com/sec/bestreviewer/integration_test_output.txt";
        final String inputFileName = "./src/test/java/com/sec/bestreviewer/integration_test_input.txt";

        String[] args = {inputFileName, outputFileName};
        File outputFile = new File(outputFileName);

        EmployeeManagement employeeManagement = new EmployeeManagement();

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    public void testLargeIntegration() throws Exception {
        final String inputFileName = "./src/test/java/com/sec/bestreviewer/large_test_input.txt";
        final String outputFileName = "./src/test/java/com/sec/bestreviewer/large_test_output.txt";

        testCreateLargeInputFile(inputFileName);

        String[] args = {inputFileName, outputFileName};
        File outputFile = new File(outputFileName);

        EmployeeManagement employeeManagement = new EmployeeManagement();

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    public void testCreateLargeInputFile(String inputFileName) {
        try {
            File file = new File(inputFileName);
            if (!file.createNewFile()) {
                return;
            }

            FileOutputStream output = new FileOutputStream(file);

            // ADD
            for (int i = 0; i < 100000; i++) {
                String inputStr = "ADD, , , ,"
                        + makeEmployeeNumber(i) + ","
                        + makeRandomName() + ","
                        + makeRandomCL() + ","
                        + makeRandomPhoneNumber() + ","
                        + makeRandomBirth() + ","
                        + makeRandomCerti() + "\n";

                output.write(inputStr.getBytes());
                output.flush();
            }
            output.close();

        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    String makeEmployeeNumber(int i) {
        Random rand = new Random();
        int year = 90 + rand.nextInt(30);

        return String.format("%02d%06d", year % 100, i + 1);
    }

    String makeRandomName() {
        Random rand = new Random();
        String[] firstNames =
                {"UAFBOWU", "TYLNXWQN", "NSESFBLRK", "VWCGM", "QSJYRB", "VLPDHI", "JYFMBMT", "KYUMOK", "GILDONG"};
        String[] LastNames = {"KIM", "LEE", "PARK", "CHOI", "AN", "OH", "HAN", "HONG"};
        return firstNames[rand.nextInt(9) ] + " " + LastNames[rand.nextInt(8)];
    }

    String makeRandomCL() {
        Random rand = new Random();
        switch (rand.nextInt(4)) {
            case 0:
                return "CL1";
            case 1:
                return "CL2";
            case 2:
                return "CL3";
            case 3:
                return "CL4";
        }
        return "CL1";
    }

    String makeRandomPhoneNumber() {
        Random rand = new Random();
        int middle = rand.nextInt(10000);
        int last = rand.nextInt(10000);

        return String.format("010-%04d-%04d", middle, last);
    }

    String makeRandomBirth() {
        Random rand = new Random();
        int year = 1970 + (rand.nextInt(30));
        int month = 1 + (rand.nextInt(12));
        int day = 1;
        if (month == 2) {
            day += rand.nextInt(28);
        } else {
            day += rand.nextInt(30);
        }

        return String.format("%04d%02d%02d", year, month, day);
    }

    String makeRandomCerti() {
        Random rand = new Random();
        switch (rand.nextInt(3)) {
            case 0:
                return "ADV";
            case 1:
                return "PRO";
            case 2:
                return "EX";
        }
        return "ADV";
    }
}
