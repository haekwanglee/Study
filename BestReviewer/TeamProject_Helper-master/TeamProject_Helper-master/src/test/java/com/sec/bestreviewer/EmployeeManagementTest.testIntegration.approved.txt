SCH,10
SCH,NONE
SCH,02712462,UAFBOWU HONG,CL3,010-9054-6560,19780825,ADV
SCH,09752663,VWCGM OH,CL3,010-6288-1811,19860720,ADV
SCH,94526445,VLPDHI LEE,CL4,010-7975-9766,19721111,EX
SCH,NONE
SCH,NONE
SCH,NONE
SCH,99398134,JYFMBMT OH,CL4,010-5269-9657,19730321,ADV
SCH,NONE
SCH,02712462,UAFBOWU HONG,CL3,010-9054-6560,19780825,ADV
SCH,NONE
SCH,NONE
SCH,NONE
SCH,02712462,UAFBOWU HONG,CL3,010-9054-6560,19780825,ADV
SCH,03717863,NSESFBLRK KIM,CL3,010-4108-4007,19770502,EX
SCH,7
SCH,NONE
SCH,93728322,QSJYRB HONG,CL4,010-3602-4064,19690603,PRO
SCH,03249872,TYLNXWQN OH,CL3,010-7910-1669,19820828,PRO
DEL,NONE
SCH,03249872,TYLNXWQN OH,CL3,010-7910-1669,19820828,PRO
MOD,03717863,NSESFBLRK KIM,CL3,010-4108-4007,19770502,EX
DEL,NONE
SCH,94526445,VLPDHI LEE,CL4,010-7975-9766,19721111,EX
MOD,NONE
DEL,NONE
SCH,93728322,QSJYRB HONG,CL4,010-3602-4064,19690603,PRO
MOD,93728322,QSJYRB HONG,CL4,010-3602-4064,19690603,PRO
SCH,09752663,VWCGM OH,CL3,010-6288-1811,19860720,ADV
SCH,03249872,TYLNXWQN OH,CL3,010-7910-1669,19820828,PRO
SCH,09752663,VWCGM OH,CL3,010-6288-1811,19860720,ADV
SCH,09752664,GG LEE,CL3,010-4108-1811,19860720,ADV
SCH,NONE
SCH,NONE
DEL,NONE
SCH,94526445,VLPDHI LEE,CL4,010-7975-9766,19721111,EX
SCH,09752663,VWCGM OH,CL3,010-6288-1811,19860720,ADV
MOD,93728322,QSJYRB HONG,CL4,010-3602-4064,19690603,PRO
DEL,93728322,QSJYRB HONG,CL4,010-3602-4064,19690603,EX
SCH,02712462,UAFBOWU HONG,CL3,010-9054-6560,19780825,ADV
SCH,02712463,UAFBOWU HONG,CL3,010-9054-6560,19780825,PRO
SCH,02712464,UAFBOWU HONG,CL3,010-9054-6560,19780825,EX
SCH,02712465,UAFBOWU HONG,CL3,010-9054-6560,19780825,ADV
SCH,02712466,UAFBOWU HONG,CL3,010-9054-6560,19780825,PRO
MOD,09752663,VWCGM OH,CL3,010-6288-1811,19860720,ADV
MOD,09752664,GG LEE,CL3,010-4108-1811,19860720,ADV
SCH,6
SCH,5
SCH,8
SCH,7
SCH,5
SCH,4
SCH,1
DEL,NONE
DEL,02712462,UAFBOWU HONG,CL3,010-9054-6560,19780825,ADV
DEL,02712463,UAFBOWU HONG,CL3,010-9054-6560,19780825,PRO
DEL,02712464,UAFBOWU HONG,CL3,010-9054-6560,19780825,EX
DEL,02712465,UAFBOWU HONG,CL3,010-9054-6560,19780825,ADV
DEL,02712466,UAFBOWU HONG,CL3,010-9054-6560,19780825,PRO
DEL,NONE
SCH,NONE
SCH,NONE
SCH,NONE
SCH,6
SCH,NONE
SCH,NONE
SCH,2
